### PR TITLE
IT3: Support lightweight GeometryTGeo

### PR DESCRIPTION
**Pull request by f3sch**
Changes included here:
- Setting detector name from extracted geometry.
- fixed a bug that the IT3 delta for pitch sizes would also be applied to ITS when built with ENABLE_UPGRADES (which is of course nonsense, since there are no use cases for building with this flag and then simulating ITS not IT3, but fixed to be self-consistent).
- When creating the aligned geometry and lightweight GeometryTGeo only do the latter for the detectors actually present in the mask.

### CODE REVIEW

**Comment by shahor02** [comment]
Note that I asked @mconcas to close the https://github.com/AliceO2Group/AliceO2/pull/12769/commits/da64d8addd853359907cdccd7f3de67fe30a6479 since using 3D field in the ITS did not show any advantage in the precision but caused ~4% slowdown.

**Comment by f3sch** [comment]
Sorry @shahor02, I do not understand your point. In #12761 da64d8a is added, no? 
I simply took the patch from the PR and applied it locally.
I do not plan to merge the three preceding commits that are attached but will rebase to whatever is dev after #12765 is merged.

**Comment by shahor02** [comment]
Ah, actually, it sneaked into my PR because I was testing 2 things at the same time, will remove it before merging.

**Comment by f3sch** [comment]
Ok understood, thanks.

File: Detectors/ITSMFT/ITS/base/src/GeometryTGeo.cxx [hunk 1]
```diff
@@ -853,19 +856,21 @@ int GeometryTGeo::extractLayerChipType(int lay) const
...
+  LOGF(info, "NLayers:%d NChips:%d\n", mNumberOfLayers, getNumberOfChips());
   for (int i = 0; i < mNumberOfLayers; i++) {
-    printf(
-      "Lr%2d\tNStav:%2d\tNChips:%2d "
-      "(%dx%-2d)\tNMod:%d\tNSubSt:%d\tNSt:%3d\tChip#:%5d:%-5d\tWrapVol:%d\n",
-      i, mNumberOfStaves[i], mNumberOfChipsPerModule[i], mNumberOfChipRowsPerModule[i],
-      mNumberOfChipRowsPerModule[i] ? mNumberOfChipsPerModule[i] / mNumberOfChipRowsPerModule[i] : 0,
-      mNumberOfModules[i], mNumberOfHalfStaves[i], mNumberOfStaves[i], getFirstChipIndex(i), getLastChipIndex(i),
-      mLayerToWrapper[i]);
+    LOGF(info,
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/base/src/GeometryTGeo.cxx:867]: Do we really need this to be logged?

File: Detectors/ITSMFT/ITS/base/src/GeometryTGeo.cxx [hunk 2]
```diff
@@ -853,19 +856,21 @@ int GeometryTGeo::extractLayerChipType(int lay) const
 //__________________________________________________________________________
 void GeometryTGeo::Print(Option_t*) const
 {
-  printf("NLayers:%d NChips:%d\n", mNumberOfLayers, getNumberOfChips());
   if (!isBuilt()) {
+    LOGF(info, "Geometry not built not yet!");
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/base/src/GeometryTGeo.cxx]:
```suggestion
    LOGF(info, "Geometry not built yet!");
```

**Review by mconcas — DISMISSED** [review]
ITS part almost untouched, few minor comments.

File: Detectors/ITSMFT/ITS/base/src/GeometryTGeo.cxx (hunk 1 above)
Comment (f3sch) [reply Detectors/ITSMFT/ITS/base/src/GeometryTGeo.cxx:867]: Probably not, if you prefer it. The only use case I see of this `Print()` is anyways in the IT3 digitizer, that's why I wanted to log this.

**Comment by f3sch** [comment]
reverting to draft until a resolution for [O2-4698](https://its.cern.ch/jira/browse/O2-4698) is found.

**Comment by shahor02** [comment]
Well, since you don't modify the data members, it should not be related to O2-4698.

**Comment by f3sch** [comment]
@shahor02 can this be merged? thanks.

**Pull request merged**